typescript-requiring-type-checking.js

```diff
@@ -15,23 +15,23 @@ module.exports = {
         /*
```

Comment:
Discussion for recommended rule: `'@typescript-eslint/restrict-plus-operands': 'error'`

Reply:
I saw the following failures with [restrict-plus-operands](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/restrict-plus-operands.md) that are worth discussing:
1. String formatting like 
   - `numericValue1 + numericValue2 + 'px'` 
   - `marker.value + '%` 
   - `'Can't select more than ` + number + ' items.'`
   - `urlString + '/' + idNumber`
   The rule suggests replacing these with template literals but is that sufficiently more readable that it's worth enforcing?
2. Adding two variables with `any` type that were obviously numbers. This seems like a legit failure.

I think I like this rule but since I hit a decent number of failures (>100) I thought it would be good to discuss.

Reply:
General agreement at today's meeting that this is a good rule but I'll leave this thread open in case people have more thoughts after adopting it in other repos.

Reply:
Marking as resolved as no recent discussion, we can reopen as needed.